providers/structured/provider.go

```diff
@@ -107,6 +119,8 @@ func (p *StructuredOutputLogProvider) Record(ctx context.Context, metrics map[st
 		ContextFields: log.FieldsFromContext(ctx),
 		Metrics:       metrics,
 	}
+	p.metricMutex.Lock()
+	defer p.metricMutex.Unlock()
```

Comment:
@chriswhelix Is there a way to unlock this mutex before `p.LogProvider.Record(ctx, metrics)` but also make sure it is called every time before the function exits?